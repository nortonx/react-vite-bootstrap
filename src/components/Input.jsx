import PropTypes from "prop-types";
import "../scss/Input.scss";
const Input = ({label, type, value, onChange, readOnly, placeholder, required}) => {
  return(
    <>
      <div className="col-sm-6">
        <label htmlFor="" className="form-label">{label}</label>
        <input 
          type={type} 
          value={value} 
          className="form-control input" 
          placeholder={placeholder}
          required={required}
          readOnly={readOnly}
          onChange={onChange}
          data-testid="input"
        />
      </div>
    </>
  );
};

Input.propTypes = {
  label: PropTypes.string,
  type: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
  readOnly: PropTypes.bool,
  placeholder: PropTypes.string,
  required: PropTypes.bool,
};

export default Input;