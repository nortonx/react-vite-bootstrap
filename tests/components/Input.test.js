import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

import Input from "../../src/components/Input";

describe("Input", () => {
  it("Should render and match snapshot", async () => {
    render(<Input />)
    const input = await screen.findByTestId("input");
    expect(input).toBeInTheDocument();
    expect(input).toMatchSnapshot();
  })
})