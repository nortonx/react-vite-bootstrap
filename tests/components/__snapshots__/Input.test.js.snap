// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Should render and match snapshot 1`] = `
<input
  class="form-control input"
  data-testid="input"
  value=""
/>
`;
